libraries/standard/mqtt/src/iot_mqtt_subscription.c

```diff
@@ -203,10 +272,56 @@ static bool _topicMatch( const IotLink_t * pSubscriptionLink,
     if( ( nameIndex == topicNameLength ) && ( filterIndex == topicFilterLength ) )
     {
         status = true;
+    }
+
+cleanup:
+
+    return status;
+}
+
+/*-----------------------------------------------------------*/
+
+static bool _topicMatch( const IotLink_t * pSubscriptionLink,
+                         void * pMatch )
+{
+    bool status = false;
+
+    /* Because this function is called from a container function, the given link
+     * must never be NULL. */
+    IotMqtt_Assert( pSubscriptionLink != NULL );
+
+    _mqttSubscription_t * pSubscription = IotLink_Container( _mqttSubscription_t,
+                                                             pSubscriptionLink,
+                                                             link );
+    _topicMatchParams_t * pParam = ( _topicMatchParams_t * ) pMatch;
+
+    /* Extract the relevant strings and lengths from parameters. */
```

Comment:
Will we run afoul of any older embedded compilers by placing these variable declarations mid-block?

Reply:
I believe this should still qualify as the top of the block?  The only lines separating it from the other declarations are comments.